runbot_travis2docker/models/runbot_branch.py

```diff
@@ -3,20 +3,75 @@
 #   Coded by: moylop260@vauxoo.com
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
+from datetime import datetime
+import requests
+
+from urlparse import urlparse
+
 from openerp import fields, models, api
 
 
 class RunbotBranch(models.Model):
     _inherit = "runbot.branch"
 
     uses_weblate = fields.Boolean(help='Synchronize with Weblate')
+    updated_weblate = fields.Datetime(help='Last update of weblate')
 
     @api.model
     def cron_weblate(self):
         for branch in self.search([('uses_weblate', '=', True)]):
-            self.env['runbot.build'].create({'branch_id': branch.id,
-                                             'name': 'HEAD',
-                                             'uses_weblate': True})
+            if (not branch.repo_id.weblate_token or
+                    not branch.repo_id.weblate_url):
+                continue
+            current_date = (datetime.strptime(branch.updated_weblate,
+                                              '%Y-%m-%d %H:%M:%S')
+                            if branch.updated_weblate
+                            else None)
+            new_date = None
+            url = branch.repo_id.weblate_url
+            session = requests.Session()
+            session.headers.update({
+                'Accept': 'application/json',
+                'User-Agent': 'runbot_travis2docker',
+                'Authorization': 'Token %s' % branch.repo_id.weblate_token
+            })
+            projects = session.get(url + '/projects/').json()
+            for project in projects['results']:
+                components = session.get('%s/projects/%s/components'
+                                         % (url, project['slug'])).json()
+                for component in components['results']:
+                    slug = branch.repo_id.name
+                    repo = branch.repo_id.name
+                    if '@' in repo:
+                        slug = repo.split('@')[1:].pop().replace('/', '-')
```

Comment:
What about use a field computed store=True from runbot.repo in order to avoid compute this value each time?
and use regex, please
E.g. https://github.com/Vauxoo/travis2docker/blob/master/src/travis2docker/git_run.py#L32-L37